tests/conftest.py

```diff
@@ -124,3 +136,19 @@ def ab_slots(tmp_path_factory: pytest.TempPathFactory) -> SlotMeta:
         slot_a_boot_dev=str(slot_a_boot_dev),
         slot_b_boot_dev=str(slot_b_boot_dev),
     )
+
+
+class ThreadpoolExecutorFixtureMixin:
+    THTREADPOOL_EXECUTOR_PATCH_PATH: str
```

Comment:
This mixin is intended for common use, because we have to patch `ThreadpoolExecutor` if the code to be tested use it. Directly using `ThreadpoolExecutor` without proper tearing down(in test file or in the source) will cause `pytest` to hang and fail to exit properly.

Reply:
My question is whether the following code is enough or not.
If another configuration(=f"{cfg.OTACLIENT_STUB_MODULE_PATH}") is required, current PR is OK.
```suggestion
    THTREADPOOL_EXECUTOR_PATCH_PATH = f"{cfg.OTACLIENT_STUB_MODULE_PATH}"
```

Reply:
`ThreadpoolExecutorFixtureMixin` is implemented as a shared fixture(so located in `conftest.py`), so it is unnatural to bind it specifically to just patch `otaclient_stub`.

Reply:
P.S.: actually the issue that `ThreadpoolExecutor` hanging the pytest doesn't always occurs, I use this fixture currently only in `test_otaclient_stub.py` just because I encountered this issue there. Actually there are other test files directly using `ThreadpoolExecutor` work fine, but if the issue occurs, I will setup the patch with this mixin.

Reply:
I see, this mixin is for common, not only for ota_client_stub, but currently only ota_client_stub is using this mix-in.